### PR TITLE
added share link after submit code smells

### PR DESCRIPTION
**Pull request by rong2**

### CODE REVIEW

File: client/src/containers/ReviewCode.js [hunk 1]
```diff
@@ -57,42 +51,23 @@ class ReviewCode extends Component {
...
        
         var content = "";
-        var scoreDisplay;
 
         if (codeReview) {
             content = codeReview.content.split("\n");
             for (var i = 0; i < content.length; i++) {
+                var num = eval(i + 1);
                 content[i] = {
-                    lineNumber: i + 1,
```
Comment (rong2) [inline client/src/containers/ReviewCode.js:77]: This was a stupid mistake and did not work as expected.

File: client/src/components/ShareLinkModal.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 import { connect } from 'react-redux';
 
-class ScoreModal extends React.Component {
+class ShareLinkModal extends React.Component {
 
 	render() {
 		if (this.props.isOpen) {
```
Comment (rogerbutt) [inline client/src/components/ShareLinkModal.js:8]: i like how props control it.

File: client/src/containers/Profile.js [hunk 3]
```diff
@@ -12,6 +12,9 @@ class Profile extends Component {
 
     constructor(props) {
         super(props);
+        this.state = {
```
Comment (rogerbutt) [inline client/src/containers/Profile.js]: should probably be part of the user state not the page state

File: client/src/containers/SubmitCodeSmells.js [hunk 4]
```diff
@@ -37,11 +34,21 @@ class SubmitCodeSmells extends Component {
...
 		const { dispatch, code, codeid, userid, selectedLines } = this.props;
-        if (selectedLines.length !== 0) {
-            dispatch(submitCode(userid, code, selectedLines));
-        } else {
-            console.log("You didn't input any code smells!");
-        }
+		if (!this.state.hasSubmitted) {
+			if (selectedLines.length !== 0) {
+				dispatch(submitCode(userid, code, selectedLines));
+				this.setState({
```
Comment (rogerbutt) [inline client/src/containers/SubmitCodeSmells.js]: it's probably not guaranteed that the submit will work but you open the modal any way. Instead maybe wait for component with receive props for a successful has submitted then open the modal. But it's a bit more complicated. Up to your digression if you want to do it that way.

File: client/src/containers/Profile.js (hunk 3 above)
Comment (rong2) [inline client/src/containers/Profile.js]: This was from Hanna's merge request, I'll change it.

File: client/src/containers/SubmitCodeSmells.js (hunk 4 above)
Comment (rong2) [inline client/src/containers/SubmitCodeSmells.js]: That's a good point. I'll look into it :+1:

**Pull request merged**